### PR TITLE
Webapp2 on a streamer

### PR DESCRIPTION
**Pull request by rtasson**
This PR makes webapp 2 functional on a streamer. The commit history is a bit logical in terms of steps taken to get to this point; it might make sense to review this per-commit. I need to test this on a regular box still, but should there be issues there this is unlikely to get re-architected; this is the perfect time for review.

Couple of things to note:
* Because the `ZonesModal` is pretty involved in hardware that is not present on this box, I've opted to make an entirely separate `StreamerOutputModal`, which is much less involved. Given that, it changes the control flow of the app a bit; a lot of scaffolding used to support zones & groups is more or less ignored and could create awkward edge cases.
* This modal is only capable of setting one stream to one RCA output. If this is insufficient, we definitely should talk about how to rearchitect the backend to better support that (possibly using Pipewire or LMS; I don't know enough about ALSA to say it can do this too but it might!) 
* I've changed `nextAvailableSource` to use `React.getRef()` to pass this state around, and in particular allow us to get fiddly with when & where we set the "next available source" (in this case, which RCA output to emit signal on.) This was necessary because the `apply` function in `StreamsModal` hangs on to the initial reference, before the actual RCA output is selected in the next modal.
* The status API call now emits an `isStreamer` bool.

from folks who know more bout the architecture of this app: did I miss anything? are there conventions I failed to observe, or React "gotchas" I neglected? I tagged folks but no worries if you feel like this PR isn't your wheelhouse; I won't be offended if you ignore it :sunglasses:

### CODE REVIEW

**Review by linknum23 — CHANGES_REQUESTED** [review]
All of the changes on the back end side look good.  Same with the react code, however I will need input from Klayton and Jonah to identify issues here.

Looking at the front end as a user:

Changes needed:
- The "source" should be identified in the player card (on the home screen) and on the player page

Future work / Nice to have:
- showing that a source is currently in use on the source selection page
- cards seem a little small now when there is no metadata (and without the zone display), should there be a minimum size to show off the cover art?

**Comment by linknum23** [comment]
Also: Bug report: I cant change the stream on a source from the home screen. The dialog pops up and appears to work but the change doesnt happen.

**Review by jonahshader — COMMENTED** [review]
I didn't test it but most of the refactoring makes sense

File: web/src/pages/Home/Home.jsx [hunk 1]
```diff
@@ -72,23 +73,26 @@ PresetAndAdd.propTypes = {
...
 const Home = () => {
     const sources = useStatusStore((s) => s.status.sources);
+    const is_streamer = useStatusStore((s) => s.status.info.is_streamer);
     const clearSourceZones = useStatusStore((s) => s.clearSourceZones);
     const [zonesModalOpen, setZonesModalOpen] = React.useState(false);
     const [streamsModalOpen, setStreamsModalOpen] = React.useState(false);
     const [presetsModalOpen, setPresetsModalOpen] = React.useState(false);
+    const [streamerOutputModalOpen, setStreamerOutputModalOpen] = React.useState(false);
+
+    const nextAvailableSource = React.useRef(null);
```
[the remark this answers is not in the document]
Comment (rtasson) [reply web/src/pages/Home/Home.jsx:83]: `pages/Home/Home.jsx` brings in `executeApplyAction` from `components/StreamsModal/StreamsModal.jsx`. Within `StreamsModal.jsx`, there's a `let applyAction = null` that gets set when the `StreamsModal` completes... but at that time, the selected source (which in this case is a streamer output) is set to "the next available source", from `Home.jsx`. This works fine when we orchestrate connecting sources to zones in the `ZonesModal` & backend; however, on a streamer `nextAvailableSource` is actually directly a streamer output, which we intend select in the next `StreamerOutputModal`. Without this shared state, the `applyAction` has the old `nextAvailableSource`.

Agreed it's a bit of a smell, shared state with poor boundaries & all. An alternative would be to restructure the app to expect the `StreamerOutputModal` to run before `StreamsModal` on streamers and pass in our correct `nextAvailableSource` directly; if we didn't do the same with `ZonesModal` on regular Amplipis, it feels somewhat incorrect still to have disparate control flows for the app (but perhaps that's less messy than this shared state?) That is to say, if we both select Zones on Amplipis *and* StreamerOutputs on streamers first, the control flow would perhaps be cleaner and we can pass the source very directly from one modal to another. As it stands, both `ZonesModal` and `StreamerOutputModal` reach back into `StreamsModal` to call the prior-set apply action; that feels pretty intentional, given how straightforward the alternative seems to me, so I opted to re-use the existing control flow with a lil bit of shared state further up. Perhaps that intentionality is an assumption; why was this coded this way?

Does that make sense? I would love suggestions here. Toying with this UI and that other project has been the extent of my React experience, and pushes the boundaries on my Javascript knowledge, so any input in this particular case is greatly appreciated.

[the remark this answers is not in the document]
Comment (rtasson) [reply web/src/pages/Home/Home.jsx]: I think c65351f oughta satisfy "remove `useRef` & pass around state better"; lemme know if this approach is more satisfactory. (sometimes it is staring you in the face and you still don't see it :facepalm:)

File: web/src/components/PlayerCard/PlayerCard.jsx
[the remark this answers is not in the document]
Comment (rtasson) [reply web/src/components/PlayerCard/PlayerCard.jsx:1]: removed it in 909f1b5

**Comment by rtasson** [comment]
> Also: Bug report: I cant change the stream on a source from the home screen. The dialog pops up and appears to work but the change doesnt happen.

I believe the changes in c65351f fix this.

it truly oughta be good to go.

**Comment by linknum23** [comment]
Note: in the future we may want to add the ability to change the output names in settings

**Review by linknum23 — APPROVED** [review]
Sweet. Changes look great.

File: web/src/pages/Home/Home.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonahshader) [reply web/src/pages/Home/Home.jsx]: ```useRef``` was probably fine, I just wasn't very familiar with its normal use cases, but c65351f looks better 👍

**Pull request merged**